jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/CachingHttpContentFactory.java

```diff
@@ -0,0 +1,540 @@
+        }
+
+        @Override
+        public RetainableByteBuffer getBuffer()
+        {
+            return _buffer;
+        }
+
+        @Override
+        public long getLastAccessedNanos()
+        {
+            return _lastAccessed.get();
+        }
+
+        @Override
+        public void setLastAccessedNanos(long nanosTime)
+        {
+            _lastAccessed.set(nanosTime);
+        }
+
+        @Override
+        public String getKey()
+        {
+            return _cacheKey;
+        }
+
+        @Override
+        public void release()
+        {
+            if (_buffer != null)
+                _buffer.release();
+        }
+
+        @Override
+        public Set<CompressedContentFormat> getPreCompressedContentFormats()
+        {
+            return _compressedFormats;
+        }
+
+        @Override
+        public HttpField getETag()
+        {
+            return _etagField;
+        }
+
+        @Override
+        public String getETagValue()
+        {
+            return _etagField == null ? null : _etagField.getValue();
+        }
+
+        @Override
+        public String getCharacterEncoding()
+        {
+            return _characterEncoding;
+        }
+
+        @Override
+        public MimeTypes.Type getMimeType()
+        {
+            return _mimeType;
+        }
+
+        @Override
+        public HttpField getContentLength()
+        {
+            return _contentLength;
+        }
+
+        @Override
+        public Instant getLastModifiedInstant()
+        {
+            return _lastModifiedInstant;
+        }
+
+        @Override
+        public HttpField getLastModified()
+        {
+            return _lastModified;
+        }
+
+        @Override
+        public String getLastModifiedValue()
+        {
+            return _lastModifiedValue;
+        }
+
+        @Override
+        public boolean isValid()
+        {
+            return true;
+        }
+    }
+
+    protected static class NotFoundHttpContent implements CachingHttpContent
+    {
+        private final AtomicLong _lastAccessed = new AtomicLong();
+
+        private final String _key;
+
+        public NotFoundHttpContent(String key)
+        {
+            _key = key;
+            _lastAccessed.set(NanoTime.now());
+        }
+
+        @Override
+        public String getKey()
+        {
+            return _key;
+        }
+
+        @Override
+        public long getLastAccessedNanos()
+        {
+            return _lastAccessed.get();
+        }
+
+        @Override
+        public void setLastAccessedNanos(long nanosTime)
+        {
+            _lastAccessed.set(nanosTime);
+        }
+
+        @Override
+        public HttpField getContentType()
+        {
+            return null;
+        }
+
+        @Override
+        public String getContentTypeValue()
+        {
+            return null;
+        }
+
+        @Override
+        public String getCharacterEncoding()
+        {
+            return null;
+        }
+
```

Comment:
I think these should all be default impls on the interface.

Reply:
and most of the other methods in this class

Reply:
Why? this is the only implementation which ever returns null and it is only an internal implementation which is never returned by `getContent()`.

Reply:
Because:
 + in future there may be other implementations.
 + this is an impl that actually does something, so best to not crowd the code with boiler plate.  Having the null impl in the interface is a better place that crowding this file.